### PR TITLE
Stake pool registry format updates

### PR DESCRIPTION
**Pull request by rvl**
Relates to #1076

# Overview

This starts updating the types and tests for the new stake pool registry format, which is indexed by pool owner(s) rather than pool id.

### CODE REVIEW

File: lib/core/test/unit/Cardano/Pool/MetadataSpec.hs [hunk 1]
```diff
@@ -118,22 +118,34 @@ dataDir = $( TH.LitE . TH.StringL <$>
...
+        , owner = unsafeFromText "ed25519_pk1afhcpw2tg7nr2m3wr4x8jaa4dv7d09gnv27kwfxpjyvukwxs8qdqwg85xp"
+        , name = "First stake pool"
+        , description = Just "It's better than SND"
+        , pledgeAddress = "addr15vz9yc5c3upgze8tg5kd7kkzxqgqfxk5a3kudp22hdg0l2za00sq2ufkk7"
+        }
+    , StakePoolMetadata
+        { ticker = unsafeFromText "TICK"
+        , homepage = "https://12345"
+        , owner = unsafeFromText "ed25519_pk1z4vh8gva25w07x8574uujuveu8gz43fu6qfln3t4prcavrvcphjsk0pdqs"
+        , name = "Pooley Mc-Poolface"
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Pool/MetadataSpec.hs:147]: **[question]**

What do you think about testing some _weird_ character sets in the name and/or description? Japanese and Chinese Kanjis, Cyrilic, Emojis and all sort of non-latin characters ?

Comment (rvl) [reply lib/core/test/unit/Cardano/Pool/MetadataSpec.hs:147]: I think this is a good idea and we should do this through the QuickCheck generators.
My approach is to "ground" the tests using a fairly simple golden example, which is an actual zip file produced with the registry validator tool*.
Then attempt to test all the corner cases with randomly generated data.

(* however we need to wait for an update of the registry tool before we can download a proper example zip - the current one in this PR is synthetic).

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 2]
```diff
@@ -517,6 +523,49 @@ instance FromText PoolId where
...
+                <> " but got "
+                <> show hrp
+        Right (_, Nothing) ->
+                Left $ TextDecodingError "Stake pool owner is invalid"
+
+instance FromJSON PoolOwner where
+    parseJSON = parseJSON >=> either (fail . show . ShowFmt) pure . fromText
+
+instance ToJSON PoolOwner where
+    toJSON = toJSON . toText
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs:567]: **[question]**

It now seems a bit inconsistent that we enforce some quite strong validations on the `PoolOwner` format, whereas we still do not validate other fields (like addresses) for instance. So, what is the rationale here :thinking: ? Shall we also enforce stricter validation for addresses there?

**Review by KtorZ — APPROVED** [review]
LGTM. But I wonder if we shouldn't be more consistent regarding the metadata validation?

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs (hunk 2 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs:567]: The `PoolOwner` is also going to be extracted from the registration certificate. It's no longer just a field of `StakePoolMetadata`, so that's why.

Comment (KtorZ) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs:567]: Got it :+1:
